packages/light-api/src/ReactiveGraph.ts

```diff
@@ -0,0 +1,41 @@
+// Copyright 2017-2018 @polkadot/light-api authors & contributors
+// This software may be modified and distributed under the terms
+// of the Apache-2.0 license. See the LICENSE file for details.
+
+import { Graph } from '@dagrejs/graphlib';
+import { assert } from '@polkadot/util';
+import { merge } from 'rxjs';
+
+export class ReactiveGraph extends Graph {
+  setReactiveEdge (v: string, w: string, label: any = [], name?: string): ReactiveGraph {
+    this.setEdge(v, w, label, name);
+
+    // Get all incoming edges into w
+    const edges = this.inEdges(w);
+
+    if (!edges) {
+      throw new Error(`Incoming edges into ${w} are empty.`);
```

Comment:
how can the incoming edges of `w` be empty considering we added an edge from `v` to `w` just before?

Reply:
It's just a TS thing, `edges` type is `Edge[] | void`, with this check, I can make sure that `edges.map` doesn't throw a TS error. Will add a "dead code" mention in the error message.